pkg/sql/catalog/descpb/structured.proto

```diff
@@ -148,6 +148,17 @@ message ColumnDescriptor {
   // pg_catalog.pg_attribute. A column cannot be both hidden and inaccessible.
   optional bool inaccessible = 17 [(gogoproto.nullable) = false];
 
+  // GeneratedAsIdentityType is a enum that represents how the creation of the
+  // column is associated with the GENERATED ... AS IDENTITY token.
+  // If the column is created with GENERATED ALWAYS AS IDENTITY token,
+  // GeneratedAsIdentityType for this column will be set to GENERATED_ALWAYS.
+  // If the column is created with GENERATED BY DEFAULT AS IDENTITY token,
+  // GeneratedAsIdentityType for this column will be set to GENERATED_BY_DEFAULT.
+  // If the column is created without using GENERATED ... AS IDENTITY token,
+  // GeneratedAsIdentityType for this column will be set to the default
+  // NOT_IDENTITY_COLUMN.
+  optional GeneratedAsIdentityType generated_as_id_type = 18 [(gogoproto.nullable) = false];
```

Comment:
`id` is a bit overloaded and might lead to confusion. May I recommend renaming to `generated_as_ident_type` if `generated_as_identity_type` is not possible?